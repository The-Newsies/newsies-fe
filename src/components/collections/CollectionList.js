import React from 'react';
import PropTypes from 'prop-types';
import CollectionItem from './CollectionItem';
import styles from '../news/NewsList.css';

export default function CollectionList({ collections, handleDelete }) {
  const collectionList = collections.map(({
    name,
    description,
    _id
  }) => (
    <li key={_id}>
      <CollectionItem 
        name={name}
        description={description}
        _id={_id}
        handleDelete={handleDelete}
      />
    </li>
  ));

  return (
    <ul className={styles.container}>
      {collectionList}
    </ul>
  );
}

CollectionList.propTypes = {
  collections: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired,
  })).isRequired,
  handleDelete: PropTypes.func.isRequired,
};
