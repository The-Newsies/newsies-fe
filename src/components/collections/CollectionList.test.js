import React from 'react';
import { shallow } from 'enzyme';
import CollectionList from './CollectionList';

describe('CollectionList component', () => {
  it('renders CollectionList', () => {
    const wrapper = shallow(<CollectionList 
      collections={[{
        name: 'Title',
        description: 'description',
        _id: 'id'
      }]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
