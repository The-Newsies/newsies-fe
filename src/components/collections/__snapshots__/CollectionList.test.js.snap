// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionList component renders CollectionList 1`] = `
<ul>
  <li
    key="id"
  >
    <CollectionItem
      _id="id"
      description="description"
      name="Title"
    />
  </li>
</ul>
`;
